src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsMobileWrapperTest.java

```diff
@@ -88,9 +88,19 @@ private static void iterateNode(Node node, String fileName, String sectionName)
                 Assert.assertTrue(wrapperMessage,
                         node.getAttributes().getNamedItem("class").getNodeValue()
                                 .contains("wrapper"));
+
                 if ("table".equals(child.getNodeName())) {
```

Comment:
Sorry that this is on old code. Do we need to iterate the children of div/span/p or do we just not have them?

Reply:
I don't exactly get want you mean there.. We have to itereate over every node because some p nodes contain imgs or pre elements, also spans and div.. we iterate them by using the iterateNode()-Method recursively.. I hope I could answer this..

Reply:
> We have to itereate over every node because some p nodes contain imgs or pre elements

I see now, this is being done with `NODES_TO_WRAP`. The highlighted recursive call is only for `table` that also needs to be wrapped as well as children examined.

Reply:
the recursive call iterates through every node in the dom and checks if it is one of `NODES_TO_WRAP`. If it is not or it is a `table`, the children are also checked, because DOM children (svg, pre, img excluded) can contain `NODES_TO_WRAP` elements.